praw/models/reddit/subreddit.py

```diff
@@ -850,6 +866,26 @@ def submit(
             url = 'https://praw.readthedocs.io'
             reddit.subreddit("reddit_api_test").submit(title, url=url)
 
+        For example to submit a self post with inline media do:
+
+        .. code-block:: python
+
+            from praw.models import InlineGif, InlineImage, InlineVideo
+
+
+            gif = InlineGif("gif1", "path/to/image.gif", "optional caption")
+            image = InlineImage("image1", "path/to/image.jpg", "optional caption")
+            video = InlineVideo("video1", "path/to/video.mp4", "optional caption")
+            selftext = "Text with {gif1}, {image1}, and {video1} inline"
+            media = [gif, image, video]
+            reddit.subreddit('redditdev').submit('title', selftext=selftext, inline_media=media)
```

Comment:
I like where this feature is going. I'd like to suggest a possible change to the interface (to see what you think about it).

What I suggest is that rather than have each piece of `InlineMedia` know its own name, the names are instead assigned as keys in a dict that gets passed into `submit`. This dict would pretty much be the `text_media` dict that you create inside `submit`. With this change, the usage would look like:

```python3
gif = InlineGif("path/to/image.gif", "optional caption")
image = InlineImage("path/to/image.jpg", "optional caption")
video = InlineVideo("path/to/video.mp4", "optional caption")
selftext = "Text with {gif1}, {image1}, and {video1} inline"
media = {'gif1': gif, 'image1': image, 'video1': video}
reddit.subreddit('redditdev').submit('title', selftext=selftext, inline_media=media)
```

The reason I suggest this is that the `InlineMedia` objects don't actually use their name for anything, so it doesn't seem like something that should be managed within that class. We only use the names as a way of identifying the objects for the `submit` method, so I think passing them in as dict keys makes it clearer that they're identifiers.

Another thing that I'd suggest is possibly catching an exception that results from referencing a piece of media that wasn't passed in (for example `submit('title', selftext='My {media}', inline_media={'different name': image})`) and raising a different exception with a more specific message about the issue.

Reply:
Great idea.